cpp/src/arrow/compute/exec/util.h

```diff
@@ -280,7 +282,7 @@ class AtomicCounter {
   std::atomic<bool> complete_{false};
 };
 
-class ThreadIndexer {
+class ARROW_EXPORT ThreadIndexer {
```

Comment:
Why is this getting exported?

Reply:
I probably had some error somewhere on Windows. I can probably remove these exports and see what breaks and re-add only the necessary ones.

Reply:
https://github.com/apache/arrow/runs/6360766271?check_suite_focus=true

Reply:
Ah, it seems we are using these utilities in our unit tests.  I hadn't thought that needed export but I suppose it does.